src/DotNet.Status.Web/Controllers/AzurePipelinesController.cs

```diff
@@ -0,0 +1,375 @@
+    public class AzurePipelinesController : ControllerBase
+    {
+        private readonly GitHubClientFactory _gitHubClientFactory;
+        private readonly IOptions<BuildMonitorOptions> _options;
+        private readonly ILogger<AzurePipelinesController> _logger;
+        private readonly Lazy<AzureDevOpsClient> _clientLazy;
+        private readonly Lazy<Task<Dictionary<string, string>>> _projectMapping;
+
+        public AzurePipelinesController(
+            GitHubClientFactory gitHubClientFactory,
+            IOptions<BuildMonitorOptions> options,
+            ILogger<AzurePipelinesController> logger)
+        {
+            _gitHubClientFactory = gitHubClientFactory;
+            _options = options;
+            _logger = logger;
+            _clientLazy = new Lazy<AzureDevOpsClient>(BuildAzureDevOpsClient);
+            _projectMapping = new Lazy<Task<Dictionary<string,string>>>(GetProjectMappingInternal);
+        }
+
+        private AzureDevOpsClient Client => _clientLazy.Value;
+
+        private AzureDevOpsClient BuildAzureDevOpsClient()
+        {
+            AzurePipelinesMonitorOptions o = _options.Value.Monitor;
+            return new AzureDevOpsClient(o.BaseUrl, o.Organization, o.MaxParallelRequests, o.AccessToken);
+        }
+
+        private async Task<Dictionary<string, string>> GetProjectMappingInternal()
+        {
+            var projects = await Client.ListProjects();
+            return projects.ToDictionary(p => p.Id, p => p.Name);
+        }
+
+        private async Task<string> GetProjectNameAsync(string id)
+        {
+            Dictionary<string, string> mapping = await _projectMapping.Value;
+            return mapping.GetValueOrDefault(id);
+        }
+
+        [HttpPost]
+        [Route("build-complete")]
+        [ProducesResponseType((int) HttpStatusCode.NoContent)]
+        public async Task<IActionResult> BuildComplete(AzureDevOpsEvent<AzureDevOpsMinimalBuildResource> buildEvent)
+        {
+            _logger.LogInformation("Build complete notification for build '{buildId}', URL: {buildUrl}",
+                buildEvent.Resource.Id,
+                buildEvent.Resource.Url);
+            string projectName = await GetProjectNameAsync(buildEvent.ResourceContainers.Project.Id);
+            if (string.IsNullOrEmpty(projectName))
+            {
+                _logger.LogWarning(
+                    "Unknown project id '{projectId}' in collection '{collectionId}' reported in notification",
+                    buildEvent.ResourceContainers.Project.Id,
+                    buildEvent.ResourceContainers.Collection.Id
+                );
+                return NoContent();
+            }
+
+            _logger.LogInformation("Resolved build '{buildId}' for project '{project}, fetching build details",
+                buildEvent.Resource.Id,
+                projectName);
+            Build build = await Client.GetBuild(projectName, buildEvent.Resource.Id);
+            if (IsIgnoredReason(build))
```

Comment:
NIT: perhaps add an "|| IsIgnoredStatus(build)"

Reply:
Sure. :-)  It was useful for debugging to know which rejection was happening, but the odds of anyone ever coming in here again is... low. :-)